### PR TITLE
INT-4158: Port Java DSL for JPA module

### PR DESCRIPTION
**Pull request by artembilan**
JIRA: https://jira.spring.io/browse/INT-4158

* Add JavaDocs
* De-Boot `JpaDslTests`
* Fix typos and generics inconsistency in the `JpaOutboundGatewayFactoryBean`

### CODE REVIEW

File: spring-integration-jpa/src/test/java/org/springframework/integration/jpa/dsl/JpaDslTests.java [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+import org.springframework.transaction.PlatformTransactionManager;
+
+/**
+ * @author Artem Bilan
+ *
+ * @since 5.0
+ */
+@RunWith(SpringRunner.class)
+@DirtiesContext
+public class JpaDslTests {
```
Comment (garyrussell) [inline spring-integration-jpa/src/test/java/org/springframework/integration/jpa/dsl/JpaDslTests.java]: `Dsl` is redundant - however, I can see it's useful to have it in the class name - I haven't been renaming these tests in the modules I ported; perhaps we should be consistent either way.

**Review by garyrussell — CHANGES_REQUESTED** [review]
That's all; minor.

File: spring-integration-jpa/src/test/java/org/springframework/integration/jpa/dsl/JpaDslTests.java (hunk 1 above)
Comment (artembilan) [inline spring-integration-jpa/src/test/java/org/springframework/integration/jpa/dsl/JpaDslTests.java]: Yeah! Not a question. 
I just was a bit paranoiac that the name like `JpaTests`  doesn't say so much about the purpose of the test. Although maybe `.dsl` package does the trick...

Will rename for consistency.

File: spring-integration-jpa/src/main/java/org/springframework/integration/jpa/dsl/JpaBaseOutboundEndpointSpec.java [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+	@Override
+	protected JpaOutboundGateway doGet() {
+		if (!this.jpaParameters.isEmpty()) {
+			this.jpaExecutor.setJpaParameters(this.jpaParameters);
+		}
+
+		/*
+		We need this artificial BeanFactory to overcome JpaOutboundGatewayFactoryBean initialization.
+		The real BeanFactory will be applied later for the target JpaOutboundGateway instance.
+		*/
```
Comment (garyrussell) [inline spring-integration-jpa/src/main/java/org/springframework/integration/jpa/dsl/JpaBaseOutboundEndpointSpec.java]: Got it - ok; let's leave it like this.

**Review by garyrussell — APPROVED** [review]

**Comment by garyrussell** [comment]
Merged as 1bba73fc060faec6141e2edd957c98ad6a56615c

**Pull request closed without merging**